### DISCUSSION TITLE
strict mode: unknown keyword: "x-response-description"

### DISCUSSION THREAD

**Discussion by flrogerw** [fastify/fastify#4003] [Q&A]
As per the @fastify/swagger documentation, I am using 'x-response-description' keyword so I can avoid having the description show up in the schema.  Now when I run my tests Fastify / Ajv puke with the error: strict mode: unknown keyword: "x-response-description".  I have tried the ajv.customOptions in the Fastify config, but no luck.  Any suggestions on how I can get past this issue?

**Comment by mcollina** [comment]
I think something like the following would do:

```js
 const fastify = Fastify({
    ajv: {
      plugins: [
        function (ajv) {
          ajv.addKeyword({ keyword: 'x-examples' })
        }
      ]
    }
  })
```

**Reply by flrogerw** [reply]
Thanks so much for the response.  I put the code in place and still getting the errors.  I will be honest, I inherited this code, I am an Express person deep down inside.  I know this should work, so I am assuming it is user error..lol. Below is how we have it set up:
**USING:   4.0.2**

**offending SCHEMA:**
{
  type: "object",
  "x-response-description": "Successful user response.",
  properties: {
    id: bigInt,
    firstName: {
      type: "string",
      minLength: 2,
      maxLength: 128
    },
    lastName: {
      type: "string",
      minLength: 2,
      maxLength: 128
    },
    roleId: {
      type: "integer"
    }
  },
  required: ["id", "firstName", "lastName", "roleId"],
  examples: exampleUserResponse
};

**CODE:**
`import fastify from "fastify";
...
const fastifyConfig = {
  ajv: {
    plugins: [
       function (ajv) => {
        ajv.addKeyword({ keyword: 'x-examples' })
      }
    ]
  },
  logger: {
    level: "warn",
    transport: prettyPrint ? { target: "pino-pretty" } : undefined,
    serializers: {
      request(request) {
        return { url: request.url };
      }
    }
  }
};

const server = await fastify(Object.assign(fastifyConfig, sslConfig));`


**ERROR:**

`
req: {
      "method": "POST",
      "url": "/content",
      "hostname": "127.0.0.1:61465",
      "remoteAddress": "::ffff:127.0.0.1",
      "remotePort": 61466
    }
    res: {
      "statusCode": 500
    }
    err: {
      "type": "Error",
      "message": "strict mode: unknown keyword: \"x-response-description\"",
      "stack":
          Error: strict mode: unknown keyword: "x-response-description"
              at checkStrictMode (/Users/rogerwilliams/studysync/traverse/api/node_modules/ajv/lib/compile/util.ts:211:28)
`
Thanks again for taking the time to look at this for me.

**Reply by mcollina** [reply]
The actual keyword is:

```js
 const fastify = Fastify({
    ajv: {
      plugins: [
        function (ajv) {
          ajv.addKeyword({ keyword: 'x-response-description' })
        }
      ]
    }
  })
```

Sorry for the bad copy-paste from my end.
